src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -107,16 +108,31 @@ TxQ::FeeMetrics::update(Application& app,
         // so clamp down on limits.
         txnsExpected_ = boost::algorithm::clamp(feeLevels.size(),
             minimumTxnCount_, targetTxnCount_);
+        recentTxnCounts_.clear();
     }
     else if (feeLevels.size() > txnsExpected_ ||
         feeLevels.size() > targetTxnCount_)
     {
+        recentTxnCounts_.emplace_back(feeLevels.size());
+        while (recentTxnCounts_.size() > setup.ledgersInQueue)
+            recentTxnCounts_.pop_front();
+        auto const iter = std::max_element(recentTxnCounts_.begin(),
+            recentTxnCounts_.end());
+        BOOST_ASSERT(iter != recentTxnCounts_.end());
+        auto const next = [&]
+        {
+            if (*iter >= txnsExpected_)
+                return *iter;
+            // 90% backoff to decrease
+            // down to the actual max.
+            return (txnsExpected_ * 9 + *iter) / 10;
```

Comment:
I'm reading this as 90% of `txnsExpected_` plus 10% of "recent tx count max"...so the result will always be slightly less than`txnsExpected_`...correct?

Reply:
Correct. It allows the limit to come down gradually.

Reply:
I find the comment somewhat confusing. Can we brainstorm on some better wording?